### PR TITLE
feat: ord add metaprotocol_validity

### PR DESCRIPTION
**Pull request by yubing744**
## Summary

Summary about this PR

- Closes #issue

### CODE REVIEW

File: frameworks/bitcoin-move/sources/ord.move [hunk 1]
```diff
@@ -619,5 +628,52 @@ module bitcoin_move::ord {
...
+
+        let protocol_type = type_info::type_name<T>();
+        let validity = MetaprotocolValidity {
+            protocol_type,
+            is_valid,
+            invalid_reason,
+        };
+
+        let validity_id = MetaprotocolValidityID{};
+        object::add_object_field_with_id<Inscription, MetaprotocolValidityID, MetaprotocolValidity>(inscription_obj, validity_id, validity)
```
Comment (jolestar) [inline frameworks/bitcoin-move/sources/ord.move:650]: Why is MetaprotocolValidity an Object field? it should be a normal field?

Comment (yubing744) [reply frameworks/bitcoin-move/sources/ord.move:650]: I follow the Open-Closed Principle and try not to modify the previous logic with new features. Fixing the previous logic may cause a large number of previous test cases to fail. But it feels like it's over-designed here, so I'd better change it to a normal field to make the logic clearer.

Comment (jolestar) [reply frameworks/bitcoin-move/sources/ord.move]: I mean, we should use `object::add_field`, not `object::add_object_field`.

I think changing the Inscription struct not a good idea.

**Comment by yubing744** [comment]
@jolestar help review

**Review by jolestar — APPROVED** [review]

**Pull request merged**